### PR TITLE
Ttl read filter

### LINKED ISSUE #163: Support Time-to-live (TTL)
We should allow users to specify a TTL when creating the tag/edge schema. All expired data should not be returned

### PR DESCRIPTION
**Pull request by steppenwolfyuetong**
For vertex and edge with TTL, when reading data, if ttl_duration + ttl_col < now, the data is filtered out.

Close #163

### CODE REVIEW

**Comment by steppenwolfyuetong** [comment]
Rebase master and please review code, thx.

File: src/parser/MaintainSentences.h [hunk 1]
```diff
@@ -76,11 +59,126 @@ class ColumnSpecificationList final {
...
+private:
+    int64_t asInt() {
+        return boost::get<int64_t>(propValue_);
+    }
+
+    const std::string& asString() {
+        return boost::get<std::string>(propValue_);
+    }
+
+    bool asBool() {
```
Comment (zlcook) [inline src/parser/MaintainSentences.h]: the method seems not be used

File: src/parser/MaintainSentences.h [hunk 2]
```diff
@@ -76,11 +59,126 @@ class ColumnSpecificationList final {
 };
 
 
+class SchemaPropItem final {
+public:
+    using Value = boost::variant<int64_t, bool, std::string>;
```
Comment (zlcook) [inline src/parser/MaintainSentences.h]: `bool` seems unnecessary

File: src/parser/MaintainSentences.h [hunk 3]
```diff
@@ -76,11 +59,126 @@ class ColumnSpecificationList final {
...
+
+    enum PropType : uint8_t {
+        TTL_DURATION,
+        TTL_COL
+    };
+
+    SchemaPropItem(PropType op, int64_t val) {
+        propType_ = op;
```
Comment (zlcook) [inline src/parser/MaintainSentences.h]: Is it better to make sure that `op` must be `TTL_DURATION` here ?

File: src/parser/MaintainSentences.h [hunk 4]
```diff
@@ -76,11 +59,126 @@ class ColumnSpecificationList final {
...
+        propValue_ = val;
+    }
+
+    SchemaPropItem(PropType op, bool val) {
```
Comment (zlcook) [inline src/parser/MaintainSentences.h]: seems not be used

File: src/graph/test/SchemaTest.cpp [hunk 5]
```diff
@@ -405,5 +406,509 @@ TEST_F(SchemaTest, metaCommunication) {
...
+        cpp2::ExecutionResponse resp;
+        std::string query = "SHOW HOSTS";
+        client->execute(query, resp);
+        std::vector<uniform_tuple_t<std::string, 2>> expected{
+            {"127.0.0.1", "1000"},
+            {"127.0.0.1", "1100"},
+        };
+        ASSERT_TRUE(verifyResult(resp, expected));
+    }
+    {   cpp2::ExecutionResponse resp;
```
Comment (zlcook) [inline src/graph/test/SchemaTest.cpp]: code style

File: src/parser/MaintainSentences.h (hunk 1 above)
Comment (steppenwolfyuetong) [reply src/parser/MaintainSentences.h]: please refer to #403

File: src/graph/test/SchemaTest.cpp (hunk 5 above)
Comment (steppenwolfyuetong) [reply src/graph/test/SchemaTest.cpp]: Good catch!

**Comment by steppenwolfyuetong** [comment]
Jenkins go!

**Comment by steppenwolfyuetong** [comment]
Please code review, thx.

**Pull request closed without merging**